Github actions: doc build failure on release branch

### Summary

On https://github.com/mantidproject/mantidimaging/actions/runs/16467470532/job/46548277644 (from https://github.com/mantidproject/mantidimaging/pull/2735), the docs action workflow failed.

```
nothing to commit, working tree clean
Error: Process completed with exit code 1.
```

I think this is because `git rm -r -q -- ${dir_name}/[!0-9]* || true` line in `git rm -r -q -- ${dir_name}/[!0-9]* || true` is not doing quite the right thing on the release branch, so `if [ "$(git status --porcelain | wc -l)" -gt "0" ]` sees changes outside the `${dir_name}` directory, but the add/commit of `${dir_name}` has no changes.

In the more common case were there are changes to the docs, everything works fine. So this is not directly causing problems, other than having the workflow show as failed.

### Steps To Reproduce

<!--- Please provide detailed steps for reproducing this issue. -->

### Current Behaviour

<!--- Please provide information about the undesired current behaviour. -->

### Expected Behaviour

<!--- Explain how the program should behave once the issue has been resolved. -->

### Context

<!--- Please provide any relevant information about your system setup. Mantid Imaging version can be obtained by running `conda list mantidimaging` and system information can be obtained by running `uname -a`. -->

### Failure Logs

<!--- [Optional] Please include any relevant log snippets here. -->

### Screenshot(s)

<!--- [Optional] Include screenshots that illustrate the problem. -->

--- a/.github/actions/publish-docs/action.yml
+++ b/.github/actions/publish-docs/action.yml
@@ -29,8 +29,8 @@ runs:
         mkdir -p ${dir_name}
         cp -a ../docs/build/html/* ${dir_name}
         md5sum ${dir_name}/index.html # if no index then there must have been a problem
-        if [ "$(git status --porcelain | wc -l)" -gt "0" ] ; then
           git add ${dir_name}
+          if ! git diff --cached --quiet -- ${dir_name} ; then
           git commit -m "Update documentation"
           git push
         else

--- a/.github/actions/publish-docs/action.yml
+++ b/.github/actions/publish-docs/action.yml
@@ -29,8 +29,8 @@ runs:
         mkdir -p ${dir_name}
         cp -a ../docs/build/html/* ${dir_name}
         md5sum ${dir_name}/index.html # if no index then there must have been a problem
-          git add ${dir_name}
-          if ! git diff --cached --quiet -- ${dir_name} ; then
+        git add ${dir_name}
+        if ! git diff --cached --quiet -- ${dir_name} ; then
           git commit -m "Update documentation"
           git push
         else

--- a/docs/release_notes/next/fix-2757-docs-release-publish-empty-commit
+++ b/docs/release_notes/next/fix-2757-docs-release-publish-empty-commit
@@ -1,0 +1,1 @@
+#2757: Fix release branch documentation publishing failing when no files change in the versioned docs directory

